fusion_bench/utils/data.py

```
@@ -94,13 +94,20 @@ def __next__(self):
                 raise RuntimeError(
-                    f"Error retrieving data from data loader: {e}"
+                    f"Error retrieving data from data loader: [{type(e).__name__}]{e}"
                 ) from e
+
         # If we get here, all attempts failed
         raise RuntimeError(
             f"Failed to retrieve data from data loader after {self.max_retries} attempts. "
-            f"Last error: {last_exception}"
+            f"Last error: [{type(last_exception).__name__}]{last_exception}. "
+            + (
+                f"The data loader appears to be empty."
```

Flake8 (7.3.0)

[error] f-string is missing placeholders

(F541)